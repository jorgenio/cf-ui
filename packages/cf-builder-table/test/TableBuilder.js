import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import {
  tableActions,
  tableReducer,
  TableBuilder
} from '../../cf-builder-table/src/index';
import {
  Table,
  TableBody,
  TableRow,
  TableCell
} from '../../cf-component-table/src/index';
import felaTestContext from '../../../felaTestContext';

let store;

beforeEach(() => {
  store = createStore(
    combineReducers({
      tables: tableReducer
    }),
    applyMiddleware(thunk)
  );
});

test('should render a table', () => {
  const component = renderer.create(
    <Provider store={store}>
      {felaTestContext(
        <TableBuilder tableName="test-table" rows={[]} columns={[]} />
      )}
    </Provider>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('should render the dataset in columns', () => {
  const component = renderer.create(
    <Provider store={store}>
      {felaTestContext(
        <TableBuilder
          tableName="test-table"
          rows={[
            { id: '1', cells: { name: 'Foo', value: 'foo' } },
            { id: '2', cells: { name: 'Bar', value: 'bar' } }
          ]}
          columns={[
            {
              label: 'Name',
              cell: cells => {
                return <TableCell key="name">{cells.name}</TableCell>;
              }
            },
            {
              label: 'Value',
              cell: cells => {
                return <TableCell key="value">{cells.value}</TableCell>;
              }
            }
          ]}
        />
      )}
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should support table props', () => {
  const component = renderer.create(
    <Provider store={store}>
      {felaTestContext(
        <TableBuilder
          tableName="test-table"
          rows={[]}
          columns={[]}
          striped
          hover
          bordered
          condensed
        />
      )}
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should support row type/accent', () => {
  const component = renderer.create(
    <Provider store={store}>
      {felaTestContext(
        <TableBuilder
          tableName="test-table"
          rows={[
            { id: '1', type: 'success', cells: {} },
            { id: '2', accent: 'red', cells: {} }
          ]}
          columns={[]}
        />
      )}
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should support flashes', () => {
  const component = renderer.create(
    <Provider store={store}>
      {felaTestContext(
        <TableBuilder
          tableName="test-table"
          rows={[{ id: '1', cells: {} }, { id: '2', cells: {} }]}
          columns={[]}
        />
      )}
    </Provider>
  );
  store.dispatch(tableActions.flashRow('test-table', '1', 'success'));
  expect(component.toJSON()).toMatchSnapshot();
});
