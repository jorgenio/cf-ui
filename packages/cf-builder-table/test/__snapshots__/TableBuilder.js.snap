// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a table 1`] = `
<table
  className="cf-16x3gyk"
>
  <thead
    className="cf-1gxgayq"
  >
    <tr
      className="cf-1p0v7iz"
    />
  </thead>
  <tbody
    className="cf-tf7x23"
  />
</table>
`;

exports[`should render the dataset in columns 1`] = `
<table
  className="cf-16x3gyk"
>
  <thead
    className="cf-1gxgayq"
  >
    <tr
      className="cf-1p0v7iz"
    >
      <th
        className="cf-1avjog3"
      >
        Name
      </th>
      <th
        className="cf-1lg90rw"
      >
        Value
      </th>
    </tr>
  </thead>
  <tbody
    className="cf-tf7x23"
  >
    <tr
      className="cf-1p0v7iz"
    >
      <td
        className="cf-19t8cnp"
      >
        Foo
      </td>
      <td
        className="cf-prxjqy"
      >
        foo
      </td>
    </tr>
    <tr
      className="cf-1dqe1yv"
    >
      <td
        className="cf-19t8cnp"
      >
        Bar
      </td>
      <td
        className="cf-prxjqy"
      >
        bar
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should support flashes 1`] = `
<table
  className="cf-16x3gyk"
>
  <thead
    className="cf-1gxgayq"
  >
    <tr
      className="cf-1p0v7iz"
    />
  </thead>
  <tbody
    className="cf-tf7x23"
  >
    <tr
      className="cf-1xobv77"
    />
    <tr
      className="cf-1dqe1yv"
    />
  </tbody>
</table>
`;

exports[`should support row type/accent 1`] = `
<table
  className="cf-16x3gyk"
>
  <thead
    className="cf-1gxgayq"
  >
    <tr
      className="cf-1p0v7iz"
    />
  </thead>
  <tbody
    className="cf-tf7x23"
  >
    <tr
      className="cf-1xobv77"
    />
    <tr
      className="cf-1dqe1yv"
    />
  </tbody>
</table>
`;

exports[`should support table props 1`] = `
<table
  className="cf-96xfvr"
>
  <thead
    className="cf-1gxgayq"
  >
    <tr
      className="cf-1p0v7iz"
    />
  </thead>
  <tbody
    className="cf-tf7x23"
  />
</table>
`;
